### ISSUE TITLE
Python: JSIIMeta uses Any type instead of TypeVar variable

### ISSUE THREAD

**Issue by michelesr** [aws/jsii#4118]
### Describe the bug

To work correctly with static type checker, it should use a TypeVar.

Please check this [issue comment](https://github.com/microsoft/pyright/issues/5156#issuecomment-1559837406) for more information.

### Expected Behavior

```python
T = TypeVar("T")
class JSIIMeta:
     ...
    def __call__(cls: Type[T], *args: Any, **kwargs) -> T:
```

### Current Behavior

```python
class JSIIMeta:
     ...
    def __call__(cls: Type[Any], *args: Any, **kwargs) -> Any:
```

### Reproduction Steps

N/A

### Possible Solution

See proposed snippet

### SDK version used

N/A

### Environment details (OS name and version, etc.)

N/A

**Comment by ThomasSteinbach** [comment]
It would be nice, if someone could address this issue.
In VSCode with the latest version of Pylance installed, all the magic of typing and suggestions didn't work anymore.

<img width="277" alt="Bildschirmfoto 2023-06-26 um 10 23 14" src="https://github.com/aws/jsii/assets/1683246/bf78f90e-b0c7-40f5-8808-0ec20224110a">

**Comment by sirrus233** [comment]
Can we get some sort of ACK that this issue has been recognized? It significantly impacts the usability of CDK.

**Comment by sirrus233** [comment]
This can be closed, with the release of JSII 1.86
